### PR TITLE
added langsmith code

### PR DESCRIPTION
**Pull request by jralduaveuthey**
added the code so that the interactions of the user with stampy in https://chat.stampy.ai/ are logged in langsmith to be able to monitor how the users interact with the chatbot

### CODE REVIEW

File: api/main.py [hunk 1]
```diff
@@ -94,6 +97,32 @@ def human(id):
     # <a href=\"https://stampy.ai/?state=6207&question=What%20is%20%22superintelligence%22%3F\">
     text = re.sub(r'<a href=\\"/\?state=(\d+.*)\\">', r'<a href=\"https://aisafety.info/?state=\1\\">', r.text)
 
+    LANGCHAIN_API_KEY = os.environ["LANGCHAIN_API_KEY"]
```
Comment (mruwnik) [inline api/main.py:100]: can you move these to `env.py` and import them from there?

File: api/src/stampy_chat/chat.py [hunk 2]
```diff
@@ -18,6 +19,12 @@
 from stampy_chat.followups import StampyChain
 from stampy_chat.citations import make_example_selector
 
+from langsmith import Client
+if os.environ.get("LANGCHAIN_API_KEY"):
+    os.environ['LANGCHAIN_TRACING_V2'] = "true"
```
Comment (mruwnik) [inline api/src/stampy_chat/chat.py:24]: can these be explicitly passed into the `Client()` constructor, rather than playing about with env variable?

**Review by mruwnik — CHANGES_REQUESTED** [review]

File: api/main.py (hunk 1 above)
Comment (jralduaveuthey) [reply api/main.py]: modified it in new [commit](https://github.com/StampyAI/stampy-chat/pull/132/commits/d94abb7d9908728577beb7377a1749368d2860e1)

**Review by mruwnik — APPROVED** [review]

File: api/src/stampy_chat/chat.py (hunk 2 above)
Comment (jralduaveuthey) [reply api/src/stampy_chat/chat.py]: modified it in new [commit](https://github.com/StampyAI/stampy-chat/pull/132/commits/d94abb7d9908728577beb7377a1749368d2860e1)

Now it raises an exception cause the setup of the client must be done with a valid key. So LANGCHAIN_TRACING_V2 sould not be "true" when there is no LANGCHAIN_API_KEY cause then the invokation of the chain does not work

**Pull request merged**